packages/firestore/exp/src/api/snapshot.ts

```diff
@@ -121,15 +121,21 @@ export class QueryDocumentSnapshot<T = firestore.DocumentData>
 
 export class QuerySnapshot<T = firestore.DocumentData>
   implements firestore.QuerySnapshot<T> {
+  readonly metadata: SnapshotMetadata;
+
   private _cachedChanges?: Array<firestore.DocumentChange<T>>;
   private _cachedChangesIncludeMetadataChanges?: boolean;
 
   constructor(
     readonly _firestore: Firestore,
     readonly query: Query<T>,
-    readonly _snapshot: ViewSnapshot,
-    readonly metadata: SnapshotMetadata
-  ) {}
+    readonly _snapshot: ViewSnapshot
+  ) {
+    this.metadata = new SnapshotMetadata(
```

Comment:
The metadata can be constructed from the view snapshot.

Reply:
I noticed that we don't pass the SnapshotMetadata to the QuerySnapshot in the main SDK. QuerySnapshot is basically a view on top of the ViewSnapshot, so the metadata matches that of the ViewSnapshot.  In the ViewSnapshot, in can be arbitrarily constructed.